examples/twim-demo/src/main.rs

```diff
@@ -0,0 +1,127 @@
+        let _clocks = hal::clocks::Clocks::new(ctx.device.CLOCK).enable_ext_hfosc();
+        ctx.core.DCB.enable_trace();
+        ctx.core.DWT.enable_cycle_counter();
+        rtt_init_print!();
+
+        let p0 = Parts::new(ctx.device.P0);
+        let scl = p0.p0_30.into_floating_input().degrade();
+        let sda = p0.p0_31.into_floating_input().degrade();
+        let btn1 = p0.p0_11.into_pullup_input().degrade();
+        let btn2 = p0.p0_12.into_pullup_input().degrade();
+        let btn3 = p0.p0_24.into_pullup_input().degrade();
+        let btn4 = p0.p0_25.into_pullup_input().degrade();
+
+        let gpiote = Gpiote::new(ctx.device.GPIOTE);
+        gpiote.port().input_pin(&btn1).low();
+        gpiote.port().input_pin(&btn2).low();
+        gpiote.port().input_pin(&btn3).low();
+        gpiote.port().input_pin(&btn4).low();
+        gpiote.port().enable_interrupt();
+
+        let twim = Twim::new(ctx.device.TWIM0, Pins { scl, sda }, Frequency::K100);
+
+        init::LateResources {
+            twim,
+            gpiote,
+            btn1,
+            btn2,
+            btn3,
+            btn4,
+        }
+    }
+
+    #[idle]
+    fn idle(_: idle::Context) -> ! {
+        rprintln!("Press button 1 to READ from addr 0x1A");
+        rprintln!("Press button 2 to WRITE to addr 0x1A");
+        rprintln!("Press button 3 to READ from addr 0x1B");
+        rprintln!("Press button 4 to WRITE from addr 0x1B");
+        loop {
+            cortex_m::asm::wfi();
+        }
+    }
+
+    #[task(binds = GPIOTE, resources = [gpiote], schedule = [debounce])]
+    fn on_gpiote(ctx: on_gpiote::Context) {
+        ctx.resources.gpiote.reset_events();
+        ctx.schedule.debounce(ctx.start + 3_000_000.cycles()).ok();
+    }
+
+    #[task(resources = [twim, gpiote, btn1, btn2, btn3, btn4])]
+    fn debounce(ctx: debounce::Context) {
+        let twim = ctx.resources.twim;
+        if ctx.resources.btn1.is_low().unwrap() {
+            rprintln!("\nREAD from address 0x1A");
+            let rx_buf = &mut [0; 8][..];
+            let res = twim.read(0x1A, rx_buf);
+            rprintln!("Result: {:?}\n{:?}", res, rx_buf);
+        }
+        if ctx.resources.btn2.is_low().unwrap() {
+            rprintln!("\nWRITE to address 0x1A");
+            let mut tx_buf = [0u8; 8];
+            tx_buf.copy_from_slice(&[1, 2, 3, 4, 5, 6, 7, 8]);
+            let res = twim.write(0x1A, &tx_buf[..]);
+            rprintln!("Result: {:?}\n{:?}", res, tx_buf);
+        }
+        if ctx.resources.btn3.is_low().unwrap() {
+            rprintln!("\nREAD from address 0x1B");
+            let rx_buf = &mut [0; 4][..];
+            let res = twim.read(0x1B, rx_buf);
+            rprintln!("Result: {:?}\n{:?}", res, rx_buf);
+        }
+        if ctx.resources.btn4.is_low().unwrap() {
+            rprintln!("\nWRITE to address 0x1B");
+            let mut tx_buf = [0u8; 4];
+            tx_buf.copy_from_slice(&[9, 10, 11, 12]);
```

Comment:
```suggestion
            let mut tx_buf = [9, 10, 11, 12];
```

Reply:
The reason for this was to ensure that the data resides in RAM, but will change to this and also pull in the embedded_hal traits that will take care of it if needed